### PR TITLE
Addresses MYRIAD-171 and MYRIAD-190.  Bumped the version to demonstra…

### PR DESCRIPTION
**Pull request by DarinJ**
…te functionality of the build.  Need to upgrade the vagrant and docker images to 2.7.2.

WIP - Don't Merge just yet.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -53,7 +53,7 @@ subprojects {
 
     ext {
         mesosVer = "0.24.1"
-        hadoopVer = "2.7.0"
+        hadoopVer = "2.7.2"
```
Comment (yufeldman) [inline build.gradle]: We should have this as property probably so not to change build.gradle every time

File: myriad-scheduler/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/recovery/MyriadFileSystemRMStateStore.java [hunk 2]
```diff
@@ -92,12 +97,39 @@ public synchronized void storeMyriadState(StoreContext sc) throws Exception {
...
+        if (m.getName().equals("updateFile")) {
+          cachedUpdateFileMethod = m;
+          break;
+        }
+      }
+    }
+  }
+
+  protected void reflectedUpdateFile(Path outputPath, byte[] data) throws InvocationTargetException, IllegalAccessException {
+    if (cachedUpdateFileMethod == null) {
```
Comment (yufeldman) [inline myriad-scheduler/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/recovery/MyriadFileSystemRMStateStore.java]: Why do you need to check here for null - you already do it inside getCachedUpdateFileMethod() method

**Comment by yufeldman** [comment]
two minor comments, otherwise looks good. +1

File: myriad-scheduler/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/recovery/MyriadFileSystemRMStateStore.java (hunk 2 above)
Comment (DarinJ) [inline myriad-scheduler/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/recovery/MyriadFileSystemRMStateStore.java]: After looking over this again, I think I can just initialize once in `InitInternal(...)` and drop the lazy eval business.  Solves the null checks much cleaner.

File: build.gradle (hunk 1 above)
Comment (DarinJ) [inline build.gradle]: Sure

**Comment by DarinJ** [comment]
@yufeldman I think this commit addresses all your concerns.  Thanks for the review, definitely better to initialize the method at startup than mess with synchronization and lazy initialization!

**Comment by yufeldman** [comment]
LGTM. +1 (though it is not binding, since I am not a commiter)

**Comment by sdaingade** [comment]
LGTM +1

**Pull request closed without merging**